src/middlewares/authenticationMiddleware.ts

```diff
@@ -35,5 +37,22 @@ const isAuthenticated = async (
     res.status(500).json({ message: 'Internal server down' })
   }
 }
-
+export const isOwner = async (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+) => {
+  try {
+    const { id } = req.params
+    const authHeader = req.headers.authorization
+    const token = authHeader.split(' ')[1]
+    const decoded = jwt.verify(token, JWT_SECRET) as JwtPayload & { id: string }
```

Comment:
This part is already covered by the isAuthenticated middleware, you can put this as a secondary middleware and start by reading the `user` from the request